commons/src/main/scala/za/co/absa/spline/common/ImmutableProperties.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.common
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, InputStream, OutputStream}
+import java.util.function.BiFunction
+import java.util.{Properties, function}
+import java.{util => ju}
+
+object ImmutableProperties {
+  def fromStream(stream: InputStream): ImmutableProperties = new ImmutableProperties(new Properties() {
+    load(stream)
+  })
+
+  def apply(props: Properties): ImmutableProperties = {
+    val baos = new ByteArrayOutputStream
+    props.store(baos, "")
+    fromStream(new ByteArrayInputStream(baos.toByteArray))
+  }
+}
+
+class ImmutableProperties private(props: Properties) extends Properties {
+  override def setProperty(key: String, value: String): Nothing = throw new UnsupportedOperationException
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use active names for operations with side effects](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4582504)